test/functional/wallet_spend_unconfirmed.py

```diff
@@ -0,0 +1,460 @@
+        self.assert_undershoots_target(parent_tx)
+
+        # create sibling tx
+        sibling_txid = wallet.sendtoaddress(address=self.def_wallet.getnewaddress(), amount=0.9, fee_rate=100)
+        sibling_tx = wallet.gettransaction(txid=sibling_txid, verbose=True)
+        self.assert_beats_target(sibling_tx)
+
+        # spend both outputs from parent transaction
+        ancestor_aware_txid = wallet.sendtoaddress(address=self.def_wallet.getnewaddress(), amount=0.5, fee_rate=self.target_fee_rate)
+        ancestor_aware_tx = wallet.gettransaction(txid=ancestor_aware_txid, verbose=True)
+
+        self.assert_spends_only_parent(ancestor_aware_tx, parent_txid)
+
+        self.assert_beats_target(ancestor_aware_tx)
+        # Child is only paying for itself…
+        resulting_fee_rate = self.calc_fee_rate(ancestor_aware_tx)
+        assert_greater_than_or_equal(1.05 * self.target_fee_rate, resulting_fee_rate)
+        # …because sibling bumped to parent to ~50 s/vB, while our target is 30 s/vB
+        resulting_ancestry_fee_rate_sibling = self.calc_set_fee_rate([parent_tx, sibling_tx])
+        assert_greater_than_or_equal(resulting_ancestry_fee_rate_sibling, self.target_fee_rate)
+        # and our resulting "ancestry feerate" is therefore BELOW target feerate
+        resulting_ancestry_fee_rate = self.calc_set_fee_rate([parent_tx, ancestor_aware_tx])
+        assert_greater_than_or_equal(self.target_fee_rate, resulting_ancestry_fee_rate)
+
+        self.generate(self.nodes[0], 1)
+        wallet.unloadwallet()
+
+    def test_confirmed_and_unconfirmed_parent(self):
+        self.log.info("Start test with one unconfirmed and one confirmed input")
+        wallet = self.setup_and_fund_wallet("confirmed_and_unconfirmed_wallet")
+        wallet.sendtoaddress(address=wallet.getnewaddress(), amount=1, fee_rate=self.target_fee_rate)
+        self.generate(self.nodes[0], 1) # Wallet has two confirmed UTXOs of ~1BTC each
+        wallet.sendtoaddress(address=wallet.getnewaddress(), amount=0.5, fee_rate=0.5*self.target_fee_rate)
+
+        # wallet has one confirmed UTXO of 1BTC and two unconfirmed UTXOs of ~0.5BTC each
+        ancestor_aware_txid = wallet.sendtoaddress(address=self.def_wallet.getnewaddress(), amount=1.4, fee_rate=self.target_fee_rate)
+        ancestor_aware_tx = wallet.gettransaction(txid=ancestor_aware_txid, verbose=True)
+        resulting_fee_rate = self.calc_fee_rate(ancestor_aware_tx)
+        assert_greater_than_or_equal(resulting_fee_rate, self.target_fee_rate)
+
+        self.generate(self.nodes[0], 1)
+        wallet.unloadwallet()
+
+
+    def run_test(self):
+        self.log.info("Starting UnconfirmedInputTest!")
+        self.target_fee_rate = 30
+        self.def_wallet  = self.nodes[0].get_wallet_rpc(self.default_wallet_name)
+        self.generate(self.nodes[0], 110)
+
+        # Test that assumptions about meeting feerate and being able to test it hold
```

Comment:
In f2daebf8d1ac3829faa6443425cb97a763e9f780 "Bump unconfirmed parent txs to target feerate"

nit: I prefer that comments about what a test does are attached to the test implementation itself rather than the caller.